web/src/stores/app-store.ts

```diff
@@ -1,33 +1,46 @@
 // Utilities
 import { defineStore } from 'pinia'
-import { Factory } from '@/interfaces/planner/FactoryInterface'
+import { Factory, FactoryTab } from '@/interfaces/planner/FactoryInterface'
 import { ref, watch } from 'vue'
 
 export const useAppStore = defineStore('app', () => {
   // Define state using Composition API
-  const factories = ref<Factory[]>(JSON.parse(localStorage.getItem('factories') ?? '[]') as Factory[])
+  const factoryTabs = ref<FactoryTab[]>(JSON.parse(localStorage.getItem('factoryTabs') ?? '[]') as FactoryTab[])
+
+  if (factoryTabs.value.length === 0) {
+    factoryTabs.value = [
+      {
+        id: crypto.randomUUID(),
+        name: 'Default',
+        // Fill the tabs from the legacy factories array if present so no data gets lost
+        factories: JSON.parse(localStorage.getItem('factories') ?? '[]'),
+      },
+    ]
+  }
+
+  const currentFactoryTabIndex = ref(0)
+  const currentFactoryTab = computed(() => factoryTabs.value[currentFactoryTabIndex.value])
+
+  const factories = computed({
+    get () {
+      return currentFactoryTab.value.factories
+    },
+    set (value) {
+      currentFactoryTab.value.factories = value
+    },
+  })
+
   const loggedInUser = ref<string>(localStorage.getItem('loggedInUser') ?? '')
   const token = ref<string>(localStorage.getItem('token') ?? '')
   const lastSave = ref<Date>(new Date(localStorage.getItem('lastSave') ?? ''))
   const lastEdit = ref<Date>(new Date(localStorage.getItem('lastEdit') ?? ''))
 
   // Watch the factories array for changes
-  watch(factories, () => {
-    localStorage.setItem('factories', JSON.stringify(factories.value))
+  watch(factoryTabs, () => {
+    localStorage.setItem('factoryTabs', JSON.stringify(factoryTabs.value))
     setLastEdit() // Update last edit time whenever the data changes, from any source.
   }, { deep: true })
 
-  // Getters
-  const getFactory = (id: number) => {
-    return factories.value.find(factory => factory.id === id)
-  }
-  const getFactories = () => {
-    return factories.value
-  }
-  const getLastEdit = () => {
-    return lastEdit.value
-  }
```

Comment:
Please don't delete these, as they are partially used by #116 and getLastEdit will eventually get used by Auth for synchronisation checking.

Reply:
Got it!